writers_rollingfilewriter.go

```diff
@@ -138,7 +140,7 @@ var compressionTypes = map[rollingArchiveType]compressionType{
 			if err != nil {
 				return err
 			}
-			return createGzip(archiveName, tar)
+			return createGzip(archiveName, bytes.NewReader(tar))
```

Comment:
As for converting more things to readers and writers, I'm generally on board. I only call out this particular writer case, because it's the inverse of the reader case that we have opted to convert.